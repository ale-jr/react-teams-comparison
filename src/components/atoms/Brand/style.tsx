import styled from "styled-components";
import { Link } from 'react-router-dom'

const Brand = styled(Link).attrs(() => ({
    to: '/',
    children: 'Ã£cme'
}))`

    transition: all ${({ theme }) => theme.animation.shorthand};

    color: ${({ theme }) => theme.colors.primary.default};
    font-size: ${({ theme }) => theme.typography.size.lg};
    font-weight: ${({ theme }) => theme.typography.weight.bold};
    text-decoration: none;

    &:hover{
        filter: brightness(0.8);
    }
`

export default Brand