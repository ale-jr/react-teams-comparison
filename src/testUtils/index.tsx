import { render, RenderOptions, RenderResult } from '@testing-library/react'
import ThemeProvider from 'providers/ThemeProvider'
import { ReactElement } from 'react'


const Providers: React.FC<{
    children: JSX.Element | JSX.Element[]
}> = ({ children }) => <ThemeProvider>{children}</ThemeProvider>


const customRender = (ui: ReactElement, options?: Omit<RenderOptions, "queries">): RenderResult => render(ui, {
    wrapper: Providers,
    ...options
})

export * from '@testing-library/react'
export {customRender as render}