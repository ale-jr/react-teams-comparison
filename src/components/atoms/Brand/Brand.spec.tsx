import { render } from "testUtils";
import "jest-styled-components";
import Brand from "./index";

describe("Brand", () => {
  it("Should match snapshot", () => {
    const { asFragment } = render(<Brand />);
    expect(asFragment()).toMatchSnapshot();
  });
});
