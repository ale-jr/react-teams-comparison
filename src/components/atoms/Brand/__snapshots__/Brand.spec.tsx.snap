// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Brand Should match snapshot 1`] = `
<DocumentFragment>
  .c0 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-transition: all 0.2s ease-in;
  transition: all 0.2s ease-in;
  color: #2678FF;
  font-size: 1.25em;
  font-weight: 700;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:hover {
  -webkit-filter: brightness(0.8);
  filter: brightness(0.8);
}

<a
    class="c0"
    href="/"
  >
    Ã£cme
  </a>
</DocumentFragment>
`;
